kong/resty/dns/client.lua

```diff
@@ -924,8 +924,12 @@ local function lookup(qname, r_opts, dnsCacheOnly, try_list)
     end
     -- perform a sync lookup, as we have no stale data to fall back to
     try_list = try_add(try_list, qname, r_opts.qtype, "cache-miss")
+    -- while kong is exiting, we cannot use timers and hence we run all our queries without synchronization
     if noSynchronisation then
       return individualQuery(qname, r_opts, try_list)
+    elseif ngx.worker and ngx.worker.exiting() then
+      log(DEBUG, PREFIX, "DNS query not synchronized because the worker is shutting down")
+      return individualQuery(qname, r_opts, try_list)
```

Comment:
If we hit this special case, it can get time consuming in environments with a large number of upstreams, even more if they are SRV records, which is getting quite common.

Kong will resolve the expired records after reloading in a more efficient way, so wdyt
```suggestion
      return nil, "will not resolve records while exiting", try_list
```

Reply:
We must be able to resolve during shutdown, so returning errors instead is not an option.